framework/include/mesh/FaceInfo.h

```diff
@@ -62,32 +63,36 @@ class FaceInfo
   /// Returns the unit normal vector for the face oriented outward from the face's elem element.
   const Point & normal() const { return _normal; }
 
-  /// Returns true if this face resides on the mesh boundary.
-  bool isBoundary() const { return (_neighbor == nullptr); }
-
   /// Returns the coordinates of the face centroid.
   const Point & faceCentroid() const { return _face_centroid; }
 
   ///@{
-  /// Returns the coordinates of the approximate face centroid
-  /// (intersection of the face and the line between the cell centroids)
-  /// in case of skewed meshes.
-  const Point & rIntersection() const { return _r_intersection; }
+  /// Returns the skewness-correction vector (vector between the approximate and real face
+  /// centroids).
+  const Point skewnessCorrectionVector() const
+  {
+    Point r_intersection =
+        _elem_info->centroid() +
+        (((_face_centroid - _elem_info->centroid()) * _normal) / (_e_cn * _normal)) * _e_cn;
+
+    return _face_centroid - r_intersection;
+  }
```

Comment:
Move this definition outside the class definition so that users can peruse APIs more quickly